kore/src/Kore/Builtin/Bool.hs

```diff
@@ -207,7 +213,34 @@ termAndEquals unifyChildren a b =
         unification2 <- unifyChildren' termLike1 operand2
         let conditions = unification1 <> unification2
         pure (Pattern.withCondition termLike1 conditions)
+    worker _ _ = empty
 
+termNotBool
+    :: forall variable unifier
+    .  InternalVariable variable
+    => MonadUnify unifier
+    => TermSimplifier variable unifier
+    -> TermLike variable
+    -> TermLike variable
+    -> MaybeT unifier (Pattern variable)
+termNotBool unifyChildren a b =
+    worker a b <|> worker b a
+  where
+    unifyChildren' boolTerm term =
+        Pattern.withoutTerm <$> unifyChildren (falseTerm term) term
+        <|> return (equalsCondition boolTerm term)
+    equalsCondition boolTerm term =
+        Condition.fromPredicate
+        $ makeEqualsPredicate_ (falseTerm boolTerm) term
+    falseTerm term = asInternal (termLikeSort term) False
+    worker termLike1 trueTerm
+      | Just BoolNot { operand } <- matchBoolNot termLike1
+      , isFunctionPattern termLike1
+      , Just value2 <- matchBool trueTerm
+      , value2
+      = lift $ do
+        condition <- unifyChildren' trueTerm operand
```

Comment:
This is not right. The terms are `notBool operand` and `true`, and then this unifies `operand` with `true`. It should unify `operand` with `false`. Actually, it can be a little more general: given `notBool operand` and (`true` or `false`) it should unify `operand` with (`false` or `true`).